org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/InvisibleProjectImporter.java

```diff
@@ -140,6 +141,20 @@ public static boolean loadInvisibleProject(IPath javaFile, IPath rootPath, boole
 		return true;
 	}
 
+	private static boolean isPartOfMatureProject(IPath sourcePath) {
+		sourcePath = sourcePath.removeTrailingSeparator();
+		List<String> segments = Arrays.asList(sourcePath.segments());
+		int index = segments.lastIndexOf("src");
+		if (index <= 0) {
+			return false;
+		}
+
+		IPath srcPath = sourcePath.removeLastSegments(segments.size() -1 - index);
+		IPath container = srcPath.removeLastSegments(1);
+		return container.append("pom.xml").toFile().exists()
+			|| container.append("build.gradle").toFile().exists();
```

Comment:
I'm not super fan of hardcoding Maven and Gradle specific things here, I think we can improve that by decoupling Build files discovery from the InvisibleProjectImporter, but it'll do for now